packages/arb-token-bridge-ui/src/hooks/useCCTP.ts

```diff
@@ -0,0 +1,254 @@
+        `${attestationApiUrl}/attestations/${attestationHash}`
+      )
+
+      const attestationResponse: AttestationResponse = await response.json()
+      return attestationResponse
+    },
+    [attestationApiUrl]
+  )
+
+  const waitForAttestation = useCallback(
+    async (attestationHash: `0x${string}`) => {
+      while (true) {
+        const attestation = await fetchAttestation(attestationHash)
+        if (attestation.status === 'complete') {
+          return attestation.attestation
+        }
+
+        await new Promise(r => setTimeout(r, 5000))
+      }
+    },
+    [fetchAttestation]
+  )
+
+  const receiveMessage = useCallback(
+    async ({
+      messageBytes,
+      attestation
+    }: {
+      messageBytes: `0x${string}`
+      attestation: `0x${string}`
+    }) => {
+      const config = await prepareWriteContract({
+        address: messengerTransmitterContractAddress,
+        abi: messengerTransmitterAbi,
+        functionName: 'receiveMessage',
+        chainId: targetChainId,
+        args: [messageBytes, attestation]
+      })
+      return writeContract(config)
+    },
+    [messengerTransmitterContractAddress, targetChainId]
+  )
+
+  const approveUSDC = useCallback(
+    async (signer: Signer, amount: BigNumber) => {
+      const contract = ERC20__factory.connect(usdcContractAddress, signer)
+      return contract.functions.approve(tokenMessengerContractAddress, amount)
+    },
+    [usdcContractAddress, tokenMessengerContractAddress]
+  )
+
+  const approveAndDeposit = useCallback(
+    async ({
+      amount,
+      provider,
+      signer,
+      onSubmit,
+      onAllowanceTooLow
+    }: {
+      amount: string
+      provider: Provider
+      signer: Signer
+      onAllowanceTooLow?: () => Promise<boolean>
+      onSubmit: () => void
+    }) => {
+      if (!walletAddress || !usdcToken) {
+        return false
+      }
+
+      const { decimals, address } = usdcToken
+      const amountRaw = utils.parseUnits(amount, decimals)
+      const allowance = await getTokenAllowanceForSpender({
+        account: walletAddress,
+        erc20Address: address,
+        provider,
+        spender: tokenMessengerContractAddress
+      })
+
+      try {
+        if (allowance.lte(amountRaw)) {
+          const shouldContinue = await onAllowanceTooLow?.()
+          if (!shouldContinue) {
+            return
+          }
+
+          const tx = await approveUSDC(signer, amountRaw)
+          await tx.wait()
+        }
+        await depositForBurn(amountRaw)
+        onSubmit()
+      } catch (error: any) {
+        if (isUserRejectedError(error)) {
+          return
+        }
+
+        Sentry.captureException(error)
+        return errorToast(
+          `There was an error approving USDC, here is more information: ${error.message}`
+        )
+      }
+    },
+    [
+      approveUSDC,
+      depositForBurn,
+      tokenMessengerContractAddress,
+      usdcToken,
+      walletAddress
+    ]
+  )
+
+  return {
```

Comment:
should probably export `approveForBurn` and `depositForBurn` too

Reply:
done